src/libecalc/common/variables.py

```diff
@@ -65,3 +66,45 @@ def get_subset_for_timestep(self, current_timestep: datetime) -> VariablesMap:
 
     def zeros(self) -> List[float]:
         return [0.0] * len(self.time_vector)
+
+    def get_variables(self):
+        return self.variables
+
+    def get_time_vector(self):
+        return self.time_vector
+
+    def get_period(self):
+        return self.period
+
+    def get_variables_map(self):
+        return self
+
+
+T = TypeVar(
+    "T",
+    bound=Union[List[datetime], Dict[str, List[Annotated[float, Field(allow_inf_nan=False)]]], Period, VariablesMap],
+    covariant=True,
+)
```

Comment:
I don't think this does anything. Using a TypeVar would require _some_ input to variables map that decided what the type is, and that the `T` was used as a return type in the methods. Unless I misunderstood something.